cob_teleop/ros/src/cob_teleop.cpp

```diff
@@ -428,6 +455,20 @@ void CobTeleop::joy_cb(const sensor_msgs::Joy::ConstPtr &joy_msg)
     switch_mode();
   }
 
+  if (free_drive_button_ >= 0 && free_drive_button_ < (int)joy_msg->buttons.size() && joy_msg->buttons[free_drive_button_] == 1)
```

Comment:
why are you checking for `joy_msg->buttons[free_drive_button_] == 1` here and in the inside-if as well?

Reply:
nicely spotted.